.circleci/osx_install_dependencies.sh

```diff
@@ -52,6 +52,7 @@ function validate_checksum {
 
 if [ ! -f /usr/local/lib/libz3.a ] # if this file does not exists (cache was not restored), rebuild dependencies
 then
+  brew update && brew upgrade
```

Comment:
```suggestion
  brew update
  brew upgrade
```

The script already has `set -e`.

But also, IIRC in the past `brew update` was enough to fix such issues. Worth a quick try since removing `upgrade` might decrease the running time a little bit.

Reply:
Sadly it is not, I tried it first. See: https://app.circleci.com/pipelines/github/ethereum/solidity/31177/workflows/10fe9641-0870-4347-a1a6-e84f002b10b4/jobs/1387681

Reply:
Ah, ok.